src/electionguard-ui/ElectionGuard.UI/ViewModels/ElectionViewModel.cs

```diff
@@ -142,36 +142,41 @@ public partial class ElectionViewModel : BaseViewModel
         }
     }
 
+    private async Task RefreshUploads(string electionId)
+    {
+        var uploads = await _uploadService.GetByElectionIdAsync(electionId);
+        BallotUploads.Clear();
+        BallotCountTotal = 0;
+        BallotAddedTotal = 0;
+        BallotSpoiledTotal = 0;
+        BallotChallengedTotal = 0;
+        BallotDuplicateTotal = 0;
+        BallotRejectedTotal = 0;
+        uploads.ForEach((upload) =>
+        {
+            BallotUploads.Add(upload);
+            BallotCountTotal += upload.BallotCount;
+            BallotAddedTotal += upload.BallotImported;
+            BallotChallengedTotal += upload.BallotChallenged;
+            BallotSpoiledTotal += upload.BallotSpoiled;
+            BallotDuplicateTotal += upload.BallotDuplicated;
+            BallotRejectedTotal += upload.BallotRejected;
+        });
+    }
+
```

Comment:
Cleanup the refresh to a separate method that can be called when the challenge spoiled ballot is changed.